### ISSUE TITLE
 Row dbl-click event

### ISSUE THREAD

**Issue by mohak1990** [zippyui/react-datagrid#150]
How can I create dbl-click row to call a function?

**Comment by felixhao28** [comment]
Each click triggers onSelectionChange and you can put a interval check inside it.

```js
lastClickTime = null;
function onSelectionCheck(name, data) {
  var thisClickTime = new Date();
  if (this.state.selected == name && thisClickTime - this.lastClickTime < 300) {
    onDblClick(name, data);
    this.lastClickTime = null;
  } else {
    this.lastClickTime = lastClickTime;
  }
}
```

**Comment by radubrehar** [comment]
@mohak1990 You can use the `rowFactory` prop to hook your own listeners on rows. Using this, I think it should be trivial to accomplish your own logic.

**Comment by mohak1990** [comment]
will you please show us an example using rowFactory for dblClick in brief?

**Comment by radubrehar** [comment]
Yep. You can use `rowFactory` in 2 ways. Either you simply use it to add new props to the default row factory, in which case, you should return undefined

ex:
```jsx
const onRowDoubleClick = (rowProps) => {
   console.log(rowProps.data, rowProps.index)
}

const rowFactory = (rowProps) => {
   rowProps.onDoubleClick = onRowDoubleClick.bind(null, rowProps)
}
<DataGrid rowFactory={rowFactory} .... />
```

or just return a valid React Element to altogether replace the default row rendering. This is probably not what you want by default, since it could prove quite tedious.

**Comment by mohak1990** [comment]
worked.. thanks a lot..

**Comment by radubrehar** [comment]
yep, certainly.

You could use `column.render` which is called with 3 args:
* value
* data
* cellProps: { rowIndex, index, style, className} - where rowIndex is the index of the row currently rendering.

If you know which row you double clicked (using the rowFactory above), it's just a matter of using `column.render` and rendering an input for the given row on that column. You could render the input with `autoFocus` (or any other well written editor - see https://github.com/zippyui/react-combo for example) and then onBlur, just remove the editor and let the grid go throw the normal rendering.

**Comment by radubrehar** [comment]
@mohak1990 looking forward to news about this, looks really promising.

**Comment by mohak1990** [comment]
I also need dropdown editor to get displayed after the last row to select a new item on the click on new button.. That ddl should be temporary..i.e. on the selection of an item from the ddl and on the click of save button, the data will get saved in the database.. If i select a data from the ddl and refresh, the update will not happen..

**Comment by mohak1990** [comment]
how can i fill that very column row on dblClick with <input>??

**Comment by dhassouni** [comment]
Is there any way to get the associated event as a pass through, if for instance, I needed event.pageX and event.pageY on my double click?
